File: src/main/resources/static/employee/services/navigationRequestService.js
/*
==========================================================
 GarageOS Navigation Request Service
 Manager
==========================================================
*/

window.NavigationRequestService = {

    async getGarageRequests(garageId) {

        return Api.get(
            `/navigation/requests/garage/${garageId}`
        );

    },


    async getRequest(requestId) {

        return Api.get(
            `/navigation/requests/${requestId}`
        );

    },


    async assignDriver(
        navigationRequestId,
        driverId
    ) {

        return Api.post(
            "/navigation/trips",
            {
                navigationRequestId:
                    navigationRequestId,

                driverId:
                    driverId
            }
        );

    }

};

File: src/main/java/com/garageos/modules/navigation/service/NavigationRequestService.java
package com.garageos.modules.navigation.service;

import com.garageos.modules.navigation.dto.request.CreateNavigationRequest;
import com.garageos.modules.navigation.dto.response.NavigationRequestResponse;

import java.util.List;

public interface NavigationRequestService {

    NavigationRequestResponse createRequest(
            Long customerId,
            CreateNavigationRequest request
    );

    NavigationRequestResponse getRequest(
            Long requestId
    );

    List<NavigationRequestResponse> getCustomerRequests(
            Long customerId
    );

    List<NavigationRequestResponse> getGarageRequests(
            Long garageId
    );
}